src/containerapp/azext_containerapp/containerapp_job_decorator.py

```diff
@@ -500,8 +514,190 @@ def update(self):
+        elif assign_system_identity:
+            identity_def["type"] = "SystemAssigned"
+        elif assign_user_identities:
+            identity_def["type"] = "UserAssigned"
+
+        if assign_user_identities:
+            identity_def["userAssignedIdentities"] = {}
+            subscription_id = get_subscription_id(self.cmd.cli_ctx)
+
+            for r in assign_user_identities:
+                r = _ensure_identity_resource_id(subscription_id, self.get_argument_resource_group_name(), r)
+                identity_def["userAssignedIdentities"][r] = {}  # pylint: disable=unsupported-assignment-operation
+
+        resources_def = None
+        if self.get_argument_cpu() is not None or self.get_argument_memory() is not None:
+            resources_def = ContainerResourcesModel
+            resources_def["cpu"] = self.get_argument_cpu()
+            resources_def["memory"] = self.get_argument_memory()
+
+        container_def = ContainerModel
+        container_def["name"] = self.get_argument_container_name() if self.get_argument_container_name() else self.get_argument_name()
+        container_def["image"] = self.get_argument_image() if not is_registry_msi_system(self.get_argument_registry_identity()) else HELLO_WORLD_IMAGE
+        if self.get_argument_env_vars() is not None:
+            container_def["env"] = parse_env_var_flags(self.get_argument_env_vars())
+        if self.get_argument_startup_command() is not None:
+            container_def["command"] = self.get_argument_startup_command()
+        if self.get_argument_args() is not None:
+            container_def["args"] = self.get_argument_args()
+        if resources_def is not None:
+            container_def["resources"] = resources_def
+
+        template_def = JobTemplateModel
+        template_def["containers"] = [container_def]
+
+        self.containerappjob_def["location"] = location
+        self.containerappjob_def["identity"] = identity_def
+        self.containerappjob_def["properties"]["environmentId"] = self.get_argument_managed_env()
+        self.containerappjob_def["properties"]["configuration"] = config_def
+        self.containerappjob_def["properties"]["template"] = template_def
+        self.containerappjob_def["tags"] = self.get_argument_tags()
+
+        if self.get_argument_workload_profile_name():
+            self.containerappjob_def["properties"]["workloadProfileName"] = self.get_argument_workload_profile_name()
+            ensure_workload_profile_supported(self.cmd, managed_env_name, managed_env_rg, self.get_argument_workload_profile_name(),
+                                              managed_env_info)
+
+        if self.get_argument_registry_identity():
+            if is_registry_msi_system(self.get_argument_registry_identity()):
+                set_managed_identity(self.cmd, self.get_argument_resource_group_name(), self.containerapp_def, system_assigned=True)
+            elif is_valid_resource_id(self.get_argument_registry_identity()):
```

Comment:
`elif is_valid_resource_id(self.get_argument_registry_identity()):`?

If the identity value is `system-environment`, how it run into line 696?